paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheet.kt

```diff
@@ -80,7 +87,53 @@ class PaymentSheet internal constructor(
         setupIntentClientSecret: String,
         configuration: Configuration? = null
     ) {
-        paymentSheetLauncher.presentWithSetupIntent(setupIntentClientSecret, configuration)
+        paymentSheetLauncher.present(
+            mode = InitializationMode.SetupIntent(setupIntentClientSecret),
+            configuration = configuration,
+        )
+    }
+
+    /**
+     * Present the payment sheet with an [InitializationMode].
+     *
+     * If the [InitializationMode] is [InitializationMode.PaymentIntent] or
+     * [InitializationMode.SetupIntent] and the underlying intent is already confirmed,
+     * [PaymentSheetResultCallback] will be invoked with [PaymentSheetResult.Completed].
+     *
+     * @param mode The [InitializationMode] to use.
+     * @param configuration An optional [PaymentSheet] configuration.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    @JvmOverloads
+    fun present(
```

Comment:
All these changes are currently restricted to the library group. Once we release this, we can deprecate the old `presentWith***Intent` methods.